packages/hyperion-autologging/src/ALElementValuePublisher.ts

```diff
@@ -4,69 +4,65 @@
 import * as ALEventIndex from "./ALEventIndex";
 import * as ALID from './ALID';
-import { ALElementTextEvent, getElementTextEvent } from './ALInteractableDOMElement';
+import { getElementTextEvent } from './ALInteractableDOMElement';
 import { ReactComponentData } from "./ALReactUtils";
-import type { ALChannelSurfaceEvent } from "./ALSurface";
+import * as ALSurface from "./ALSurface";
 import { AUTO_LOGGING_SURFACE } from "./ALSurfaceConsts";
 import * as ALSurfaceMutationPublisher from "./ALSurfaceMutationPublisher";
 import { getAncestralSurfaceNode, getSurfacePath } from "./ALSurfaceUtils";
-import { ALElementEvent, ALFlowletEvent, ALLoggableEvent, ALMetadataEvent, ALReactElementEvent, ALSharedInitOptions } from "./ALType";
-
-export type ALElementValueEventData = Readonly<
-  ALLoggableEvent &
-  ALFlowletEvent &
-  ALReactElementEvent &
-  ALElementTextEvent &
-  ALMetadataEvent &
-  ALElementEvent &
-  {
-    surface: string;
-    value: string;
-  }
->;
+import { ALElementEvent, ALSharedInitOptions } from "./ALType";
 
-export type ALChannelElementValueEvent = Readonly<{
-  al_element_value_event: [ALElementValueEventData],
-}
->;
 
-export type ALElementValueChannel = Channel<ALChannelElementValueEvent & ALChannelSurfaceEvent>;
+import * as ALUIEventPublisher from "./ALUIEventPublisher";
 
 
 export type InitOptions = Types.Options<
+  ALUIEventPublisher.InitOptions &
   ALSharedInitOptions &
   {
-    channel: ALElementValueChannel;
-    cacheElementReactInfo: boolean;
+    surfaceChannel: ALSurface.InitOptions['channel']
   }
 >;
 
 export function publish(options: InitOptions): void {
-  const { channel } = options;
+  const { channel, surfaceChannel } = options;
+
+  const changeEvent = options.uiEvents.find(config => config.eventName === 'change');
+
+  if (!changeEvent) {
+    // only enable this feature if `change` event is enabled.
+    return;
+  }
+
+  const { cacheElementReactInfo } = changeEvent;
+
+  type PartialALEventValueEventData =
+    Pick<ALUIEventPublisher.ALUIEventData, "surface" | "value" | "metadata" | "relatedEventIndex"> &
+    Pick<ALElementEvent, "element">;
 
-  type PartialALEventValueEventData = Pick<ALElementValueEventData, "surface" | "element" | "value" | "metadata" | "relatedEventIndex">;
   function emitEvent(elementValueEventData: PartialALEventValueEventData) {
     const { element, surface } = elementValueEventData;
 
-    const elementText = getElementTextEvent(element, surface);
-
     let reactComponentData: ReactComponentData | null = null;
-    if (options.cacheElementReactInfo) {
+    if (cacheElementReactInfo) {
       const elementInfo = ALElementInfo.getOrCreate(element);
       reactComponentData = elementInfo.getReactComponentData();
     }
 
+    const elementText = getElementTextEvent(element, surface);
     const flowlet = options.flowletManager.top();
 
-    channel.emit('al_element_value_event', {
+    channel.emit('al_ui_event', {
+      event: 'change',
```

Comment:
Oh interesting, so you wouldn't need to enable `eventName: 'change'`  in the UI event publisher config?